packages/jupyter-ai/jupyter_ai/personas/persona_manager.py

```diff
@@ -0,0 +1,174 @@
+
+        for persona_ep in persona_eps:
+            try:
+                persona_class = persona_ep.load()
+                assert issubclass(persona_class, BasePersona)
+                persona_classes.append(persona_class)
+                class_module, class_name = persona_ep.value.split(":")
+                self.log.info(
+                    f"  - Loaded AI persona class '{class_name}' from '{class_module}' using entry point '{persona_ep.name}'."
+                )
+            except Exception as e:
+                self.log.error(
+                    f"  - Unable to load AI persona from entry point `{persona_ep.name}` due to an exception printed below."
+                )
+                self.log.exception(e)
+                continue
+
+        if len(persona_classes) > 0:
+            elapsed_time_ms = (time_ns() - start_time_ns) // 1000
+            self.log.info(
+                f"SUCCESS: Loaded {len(persona_classes)} AI persona classes from entry points. Time elapsed: {elapsed_time_ms}ms."
+            )
+        else:
+            self.log.error(
+                "ERROR: Jupyter AI has no AI personas available. "
+                + "Please verify your server configuration and open a new issue on our GitHub repo if this warning persists."
+            )
+        PersonaManager._persona_classes = persona_classes
+
+    def _init_personas(self) -> dict[str, BasePersona]:
+        """
+        Initializes the list of persona instances for the YChat instance passed
+        to the constructor.
+        """
+        if hasattr(self, "_personas") or len(PersonaManager._persona_classes) == 0:
+            return
+        persona_classes = PersonaManager._persona_classes
+        assert isinstance(persona_classes, list)
+
+        self.log.info(
+            f"PENDING: Initializing AI personas for chat room '{self.ychat.get_id()}'..."
+        )
+        start_time_ns = time_ns()
+
+        personas: dict[str, BasePersona] = {}
+        for Persona in persona_classes:
+            persona = Persona(
+                ychat=self.ychat,
+                manager=self,
+                config=self.config_manager,
+                log=self.log,
+            )
+            if persona.id in personas:
+                class_name = persona.__class__.__name__
+                self.log.warning(
+                    f"  - WARNING: Skipping persona '{persona.name}' from '{persona.__module__}' because another persona has an identical ID '{persona.id}'. "
+                    + f"Personas must all have unique IDs. Please rename the persona class from '{class_name}' to something unique to dismiss this warning."
+                )
+                continue
+            else:
+                self.log.info(
+                    f"  - Initialized persona '{persona.name}' (ID: '{persona.id}')."
+                )
+                personas[persona.id] = persona
+
+        elapsed_time_ms = (time_ns() - start_time_ns) // 1000
```

Comment:
Should this have 1_000_000 instead of 1000?

Reply:
YES, great catch!! Thank you.